packages/hdwallet-portis/src/bitcoin.ts

```diff
@@ -0,0 +1,280 @@
+      result = payments.p2sh({
+        redeem: payments.p2wpkh(args)
+      })
+      break
+    default:
+      throw new Error(`Unsupported scriptType ${scriptType}`)
+  }
+
+  if(msg.showDisplay === true) {
+    if (!verifyScriptTypePurpose(scriptType, purpose)) {
+      throw new Error(`Invalid scriptType ${scriptType} for purpose ${purpose}`)
+    }
+
+    portis.showBitcoinWallet(addressNListToBIP32(msg.addressNList))
+  }
+
+  return result.address
+}
+
+export function verifyScriptTypePurpose(scriptType: BTCInputScriptType, purpose: number): boolean {
+  return (
+      (purpose === 0x80000000 + 44 && scriptType === BTCInputScriptType.SpendAddress ) ||
+      (purpose === 0x80000000 + 49 && scriptType === BTCInputScriptType.SpendP2SHWitness ) ||
+      (purpose === 0x80000000 + 84 && scriptType === BTCInputScriptType.SpendWitness )
+  )
+}
+
+export function legacyAccount (coin: Coin, slip44: number, accountIdx: number): BTCAccountPath {
+  return {
+    coin,
+    scriptType: BTCInputScriptType.SpendAddress,
+    addressNList: [ 0x80000000 + 44, 0x80000000 + slip44, 0x80000000 + accountIdx ]
+  }
+}
+
+export function segwitAccount (coin: Coin, slip44: number, accountIdx: number): BTCAccountPath {
+  return {
+    coin,
+    scriptType: BTCInputScriptType.SpendP2SHWitness,
+    addressNList: [ 0x80000000 + 49, 0x80000000 + slip44, 0x80000000 + accountIdx ]
+  }
+}
+
+export function segwitNativeAccount (coin: Coin, slip44: number, accountIdx: number): BTCAccountPath {
+  return {
+    coin,
+    scriptType: BTCInputScriptType.SpendWitness,
+    addressNList: [ 0x80000000 + 84, 0x80000000 + slip44, 0x80000000 + accountIdx ]
+  }
+}
+
+
+export function btcNextAccountPath (msg: BTCAccountPath): BTCAccountPath | undefined {
+  let description = describeUTXOPath(msg.addressNList, msg.coin, msg.scriptType)
+  if (!description.isKnown) {
+    return undefined
+  }
+
+  let addressNList = msg.addressNList
+
+  if (addressNList[0] === 0x80000000 + 44 ||
+      addressNList[0] === 0x80000000 + 49 ||
+      addressNList[0] === 0x80000000 + 84) {
+    addressNList[2] += 1
+    return {
+      ...msg,
+      addressNList
+    }
+  }
+
+  return undefined
+}
+
+export function btcGetAccountPaths (msg: BTCGetAccountPaths): Array<BTCAccountPath> {
+  const slip44 = slip44ByCoin(msg.coin)
+  const bip44 = legacyAccount(msg.coin, slip44, msg.accountIdx)
+  const bip49 = segwitAccount(msg.coin, slip44, msg.accountIdx)
+  const bip84 = segwitNativeAccount(msg.coin, slip44, msg.accountIdx)
+
+  let paths: Array<BTCAccountPath> = {
+    'Bitcoin':  [bip44, bip49, bip84]
+  }[msg.coin] || []
+
+  if (msg.scriptType !== undefined)
+    paths = paths.filter(path => { return path.scriptType === msg.scriptType })
+
+  return paths
+}
+
+export function btcSupportsScriptType (coin: Coin, scriptType: BTCInputScriptType): Promise<boolean> {
+
+  console.log('btcSupportsScriptType!!', {coin, scriptType})
+  if(coin !== 'Bitcoin')
+    return Promise.resolve(false)
```

Comment:
if you make the function `async`, you can just `return false` here. a little cleaner.

Reply:
k